### PR TITLE
Add `anydata` type

### PR DESCRIPTION
**Pull request by pubudu91**
```
int|float|string|boolean|byte|table|json|xml|records (with only `anydata` fields)|map<anydata>|finite types (with only `anydata` fields)|anydata[]|()
```

## Samples
```ballerina
import ballerina/io;

public function main() {
    anydata adi = 10;
    anydata adf = 23.45;
    anydata adb = true;
    anydata ads = "Hello World!";

    io:println(adi is int ? adi : -1);
    io:println(adf is float ? adf : -1);
    io:println(adb is boolean ? adb : -1);
    io:println(ads is string ? ads : -1);
}
```
The above produces the following output:
```
10
23.45
true
Hello World!
```

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java [hunk 1]
```diff
@@ -180,7 +181,39 @@ public boolean isSameType(BType source, BType target) {
...
+
+        if (type.tag == TypeTags.TUPLE) {
+            BTupleType tupleType = (BTupleType) type;
+            return allDataTypes(tupleType.tupleTypes);
+        }
+
+        return type.tag == TypeTags.ARRAY && isDataType(((BArrayType) type).eType);
+    }
+
+    private boolean allDataTypes(Collection<BType> types) {
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java]: rename to isDataType()

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java [hunk 2]
```diff
@@ -180,7 +181,39 @@ public boolean isSameType(BType source, BType target) {
     }
 
     public boolean isValueType(BType type) {
-        return type.tag < TypeTags.TYPEDESC;
+        return type.tag < TypeTags.JSON;
+    }
+
+    public boolean isDataType(BType type) {
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java]: rename to isAnydata()

**Comment by MaryamZi** [comment]
Shouldn't finite types with values from just the `anydata` value space be considered `anydata`? For example, shouldn't the following (`anydata a = s`) be allowed?

```ballerina
public type Separator ","|"\t"|":";
@final public Separator COMMA = ",";
@final public Separator TAB = "\t";
@final public Separator COLON = ":";

public function main() {
    Separator s = "\t";
    anydata a = s;
}
```

**Comment by pubudu91** [comment]
Makes sense. +1

**Pull request merged**